drivers/adc/ad7091r5/ad7091r5.c

```diff
@@ -0,0 +1,617 @@
+				uint16_t data)
+{
+	uint16_t reg_data;
+	int32_t ret;
+
+	if (!dev)
+		return -EINVAL;
+
+	ret = ad7091r5_i2c_reg_read(dev, reg_addr, &reg_data);
+	if (ret < 0)
+		return ret;
+
+	reg_data &= ~mask;
+	reg_data |= data;
+
+	return ad7091r5_i2c_reg_write(dev, reg_addr, reg_data);
+}
+
+/**
+ * Set sleep mode.
+ * @param dev - The device structure.
+ * @param mode - Sleep mode.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t ad7091r5_sleep_mode(struct ad7091r5_dev *dev,
+			    enum ad7091r5_sleep_mode mode)
+{
+	if (!dev)
+		return -EINVAL;
+
+	return ad7091r5_i2c_write_mask(dev, AD7091R5_REG_CONF,
+				       REG_CONF_SLEEP_MODE_MASK,
+				       REG_CONF_SLEEP_MODE(mode));
+}
+
+/**
+ * Set output port value.
+ * @param dev - The device structure.
+ * @param port - Port number.
+ * @param value - Value.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t ad7091r5_set_port(struct ad7091r5_dev *dev, enum ad7091r5_port port,
+			  bool value)
+{
+	uint16_t mask;
+	uint16_t val;
+
+	if (!dev)
+		return -EINVAL;
+
+	switch (port) {
+	case AD7091R5_GPO0:
+		mask = REG_CONF_GPO0_MASK;
+		val = REG_CONF_GPO0(value);
+		break;
+	case AD7091R5_GPO1:
+		mask = REG_CONF_GPO1_MASK;
+		val = REG_CONF_GPO1(value);
+		break;
+	case AD7091R5_GPO2:
+		mask = REG_CONF_GPO2_MASK;
+		val = REG_CONF_GPO2(value);
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return ad7091r5_i2c_write_mask(dev, AD7091R5_REG_CONF,
+				       mask, val);
+}
+
+/**
+ * Set GPO0 mode.
+ * @param dev - The device structure.
+ * @param mode - GPO0 new mode.
+ * @param is_cmos - 0: GPO0 is open drain
+ * 		  - 1: GPO0 is CMOS.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t ad7091r5_set_gpo0_mode(struct ad7091r5_dev *dev,
+			       enum ad7091r5_gpo0_mode mode, bool is_cmos)
+{
+	uint16_t value;
+
+	if (!dev)
+		return -EINVAL;
+
+	switch (mode) {
+	case AD7091R5_GPO0_ENABLED:
+		value = 0;
+		break;
+	case AD7091R5_GPO0_ALERT:
+		value = REG_CONF_GPO0_ALERT(1);
+		break;
+	case AD7091R5_GPO0_BUSY:
+		value = REG_CONF_GPO0_BUSY(1) | REG_CONF_GPO0_ALERT(1);
+		break;
+	default:
+		return -EINVAL;
+		break;
+	}
+	value |= REG_CONF_GPO0_DRIVE_TYPE(is_cmos);
+
+	return ad7091r5_i2c_write_mask(dev, AD7091R5_REG_CONF,
+				       REG_CONF_GPO0_MODE_MASK, value);
+}
+
+/**
+ * Set cycle timer for autocycle mode.
+ * @param dev - The device structure.
+ * @param timer - Converter sampling period.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t ad7091r5_set_cycle_timer(struct ad7091r5_dev *dev,
+				 enum ad7091r5_cycle_timer timer)
+{
+	if (!dev)
+		return -EINVAL;
+
+	return ad7091r5_i2c_write_mask(dev, AD7091R5_REG_CONF,
+				       REG_CONF_CYCLE_TIMER_MASK,
+				       REG_CONF_CYCLE_TIMER(timer));
+}
+
+/**
+ * Get alert.
+ * @param dev - The device structure.
+ * @param channel - Channel.
+ * @param alert - Alert type.
+ * @return SUCCESS in case of success, FAILURE otherwise.
```

Comment:
i can see the return codes were updated in this version, but doxy should be updated too